File: packages/sdk/android/src/main/java/io/flexn/sdk/animators/AnimatorSelector.java
package io.flexn.sdk.animators;

import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.views.view.ReactViewGroup;

import java.util.Objects;

public class AnimatorSelector {
    private static final String ANIMATION_TYPE_SCALE = "scale";
    private static final String ANIMATION_TYPE_SCALE_WITH_BORDER = "scale_with_border";
    private static final String ANIMATION_TYPE_BORDER = "border";
    private static final String ANIMATION_TYPE_FADE = "fade";
    private static final String ANIMATION_TYPE_BG_COLOR = "background";

    public static AbstractAnimator selectAnimator(ReactViewGroup view, ReadableMap args) {
        AbstractAnimator animator = null;
        String animationType = args.getString("type");

        switch (Objects.requireNonNull(animationType)) {
            case ANIMATION_TYPE_SCALE:
                animator = new ScaleAnimator(view, args);
                break;
            case ANIMATION_TYPE_BORDER:
                animator = new BorderAnimator(view, args);
                break;
            case ANIMATION_TYPE_SCALE_WITH_BORDER:
                animator = new ScaleWithBorderAnimator(view, args);
                break;
            case ANIMATION_TYPE_FADE:
                animator = new FadeAnimator(view, args);
                break;
            case ANIMATION_TYPE_BG_COLOR:
                animator = new BackgroundColorAnimator(view, args);
                break;
            default:
                break;
        }

        return animator;
    }
}

File: packages/create/ios/animators/AnimatorSelector.swift
import Foundation

class AnimatorSelector {
    enum AnimationTypes: String {
        case animation_scale
        case animation_border
        case animation_scale_with_border
    }
    
    static func selectAnimator(view: UIView, args: NSDictionary) -> Animator? {
        let animationType = args["type"] as? String
        var animator: Animator? = nil
                
        switch animationType! {
            case "scale":
                animator = ScaleAnimator(view: view, args: args)
                break
            case "border":
                animator = BorderAnimator(view: view, args: args)
                break
            case "scale_with_border":
                animator = ScaleWithBorderAnimator(view: view, args: args)
                break
            case "fade":
                animator = FadeAnimator(view: view, args: args)
                break
            case "background":
                animator = BackgroundAnimator(view: view, args: args)
                break
            default: break
        }
        
        return animator ?? nil
    }
}
